tests/utils.py

```diff
@@ -128,7 +125,7 @@ def cached_contract(state, definition, deployed):
     return contract
 
 
-class Signer():
+class ActivatedSigner():
```

Comment:
It's not clear to me what the right name for this is. I was considering moving the transaction functionality upstream to Nile, but the reason we had this split in the first time is that Nile's should be generic enough while this is tied to StarkNet's testing framework -- which is not good to send transactions to public chains.

We could either abstract the provider (testing fw / starknet.py / public net?) on Nile's side and just provision the testing fw from each test and not have this intermediate abstraction, or maybe rename this one to TestSigner or something like that.

Reply:
Hmmm yeah, I struggled a little bit with naming this (`SignerWithAccount`, `ExtendedSigner`, and this). I think renaming to `TestSigner` is perfectly fine. We're just using it here to test the contracts anyway.